backend/app/Actions/User/RegisterUserAction.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\Actions\User;
+
+use app\Repositories\Contracts\UserRepository;
+use Tymon\JWTAuth\Facades\JWTAuth;
+
+class RegisterUserAction
+{
+    private $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function execute(RegisterRequest $request): RegisterResponse
+    {
+        $user = $this->userRepository->create([
```

Comment:
you forgot change usage of repository according to method save. Create here new User model and assign properties from request